### PR TITLE
Upgrade SDKs to use the latests releases of OneSignal

### PR DESCRIPTION
**Pull request by williamrijksen**
Upgrade the module to use the latest releases of the SDKs on android and iOS.

- [Android 3.8.0](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.8.0)
- [iOS 2.7.1](https://github.com/OneSignal/OneSignal-iOS-SDK/releases/tag/2.7.1)

### CODE REVIEW

**Comment by jvandijk** [comment]
@williamrijksen We've overrided the payload setup for iOS in our own OneSignalPayload class. This has now completely been refactored in the OneSignal 2.6.x release. Have you taken this refactor into account?

**Comment by williamrijksen** [comment]
@jvandijk A functional test is still required, can you review this?

**Comment by jvandijk** [comment]
The module unfortunately crashes on the toDictionary structure... I think we have to add checks to see if data is set

**Comment by williamrijksen** [comment]
@jvandijk Fixed!

File: ios/Classes/OneSignalModuleHelper.m [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
++ (NSDictionary *)toDictionary: (OSNotificationPayload *)payload
+{
+    return @{@"notificationID": payload.notificationID ?: [NSNull null],
+             @"templateID": payload.templateID ?: [NSNull null],
+             @"templateName": payload.templateName ?: [NSNull null],
+             @"contentAvailable": @(payload.contentAvailable),
+             @"mutableContent": @(payload.mutableContent),
+             @"category": payload.category ?: [NSNull null],
+             @"badge": @(payload.badge),
```
Comment (williamrijksen) [inline ios/Classes/OneSignalModuleHelper.m:21]: missing badgeIncrement.

**Comment by chmiiller** [comment]
Guys, everything seems to work fine when merging this into Master. The only thing that is missing is the "notificationOpened" when the app is closed.
The `[self.delegate notificationOpened:[OneSignalModuleHelper toDictionary:payload]];` works and I'm printing the payload casted as a dictionary and everything seems fine, but the payload is never sent to `self.delegate notificationOpened` do you have any idea why?
Thanks!

**Comment by jvandijk** [comment]
Closing in favor of #77 and #79

**Pull request closed without merging**